consensus/istanbul/backend/handler.go

```diff
@@ -346,6 +346,8 @@ func (sb *Backend) newChainHead(newBlock *types.Block) {
 
 		sb.logger.Info("Validator Election Results", "address", sb.ValidatorAddress(), "elected", valSetIndex >= 0, "number", newBlock.Number().Uint64())
 
+		sb.announceMu.Lock()
+		defer sb.announceMu.Unlock()
 		if sb.announceRunning {
```

Comment:
Mind adding a comment on why is this necessary? Is it for protecting the access of the announceRunning variable, or because of the RefreshValPeers?

Reply:
Done